### DISCUSSION TITLE
Smarthings Continously Sending STATUS reports to Tasmota

### DISCUSSION THREAD

**Discussion by jamescatania** [arendst/Tasmota#15360] [Q&A] [answered]
### PROBLEM DESCRIPTION
I am aware that this might be more of smartthigns issue rather than tasmota, but my system is extremely non-responsive and ST is sending all my 60+ tasmota devices continous STATUS commands

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): Sonoff Basic
- [X] Tasmota binary firmware version number used: _____
  - [X] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [X] Provide the output of command: `Backlog Template; Module; GPIO 255`: 
```lua
  Configuration output here:
07:37:13.528 RSL: RESULT = {"NAME":"Sonoff Basic","GPIO":[32,1,1,1,1,0,0,0,224,320,1,0,0,0],"FLAG":0,"BASE":1}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [X] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
STATUS = {"Status":{"Module":1,"DeviceName":"TasmotaOffice","FriendlyName":["TasmotaOffice"],"Topic":"tasmota_C56473","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}

```
- [X] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
07:39:39.660 SRC: WebCommand from 192.168.1.23
07:39:39.663 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '8'
07:39:39.669 RSL: STATUS8 = {"StatusSNS":{"Time":"2022-04-13T07:39:39"}}
07:39:51.103 WIF: Checking connection...
07:39:53.564 WIF: Sending Gratuitous ARP
07:40:11.108 WIF: Checking connection...
07:40:31.094 WIF: Checking connection...
07:40:39.966 HTP: Command
07:40:39.972 SRC: WebCommand from 192.168.1.23
07:40:39.974 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '8'
07:40:39.979 RSL: STATUS8 = {"StatusSNS":{"Time":"2022-04-13T07:40:39"}}
07:40:40.281 HTP: Command
07:40:40.287 SRC: WebCommand from 192.168.1.23
07:40:40.290 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 2, Data '11'
07:40:40.296 RSL: STATUS11 = {"StatusSTS":{"Time":"2022-04-13T07:40:40","Uptime":"0T15:29:48","UptimeSec":55788,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"ICARUS","BSSId":"74:AC:B9:D3:59:57","Channel":11,"Mode":"11n","RSSI":100,"Signal":-32,"LinkCount":1,"Downtime":"0T00:00:05"}}}
07:40:51.101 WIF: Checking connection...
07:40:53.559 WIF: Sending Gratuitous ARP
07:41:07.546 RSL: STATE = {"Time":"2022-04-13T07:41:07","Uptime":"0T15:30:15","UptimeSec":55815,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"ICARUS","BSSId":"74:AC:B9:D3:59:57","Channel":11,"Mode":"11n","RSSI":100,"Signal":-31,"LinkCount":1,"Downtime":"0T00:00:05"}}
07:41:11.096 WIF: Checking connection...
07:41:31.080 WIF: Checking connection...
07:41:36.899 HTP: Command
07:41:36.905 SRC: WebCommand from 192.168.1.23
07:41:36.908 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 1, Data '8'
07:41:36.913 RSL: STATUS8 = {"StatusSNS":{"Time":"2022-04-13T07:41:36"}}
07:41:38.110 HTP: Command
07:41:38.116 SRC: WebCommand from 192.168.1.23
07:41:38.118 CMD: Grp 0, Cmnd 'STATUS', Idx 1, Len 2, Data '11'
07:41:38.124 RSL: STATUS11 = {"StatusSTS":{"Time":"2022-04-13T07:41:38","Uptime":"0T15:30:46","UptimeSec":55846,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"ICARUS","BSSId":"74:AC:B9:D3:59:57","Channel":11,"Mode":"11n","RSSI":100,"Signal":-32,"LinkCount":1,"Downtime":"0T00:00:05"}}}
07:41:51.086 WIF: Checking connection...
07:41:53.541 WIF: Sending Gratuitous ARP

```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by ascillato** [comment]
What do you want Tasmota to do? From your logs, Tasmota is just answering what is being asked by HTTP.

**Reply by sfromis** [reply]
Talk to whoever made that primitive Tasmota integration for Smartthings. A much better way of doing it would be to use MQTT, where Tasmota can automatically report status changes as it happens, no polling needed. And Tasmota is not optimized for the slower http requests anyway.

More generally, the extra power draw from constant commands to be processed may overtax the 3.3V power supply ability of some borderline devices, leading to them becoming "less responsive".

**Reply by joba-1** [reply]
As sfromis said: polling is a problem and likely the cause for unresponsiveness and increasing rapidly with the number of devices.

From its Readme I think a software like https://github.com/sgupta999/mqtt-bridge-smartthings can help you to only send info when it changes. In the long run I'd change to something where I have full control over what's happening and not relying on workarounds

**Reply by sfromis** [reply]
In case you're using "tasmota-connect", this thread may be related: https://github.com/hongtat/tasmota-connect/issues/22

**Reply by jamescatania** [reply]
I am indeed using tasmota-connect. and it is indeed the issue with Smartthings as suspected. As for MQTT.

@joba-1 What would you recommend, something like Home Assistant? As it has been on my mind for a bit that I'm not happy with smartthings.  I woudl really be interested

@sfromis That jsut says that the Tasmota-connect for ST does not have an MQTT client. 

![IMG_20210531_162814](https://user-images.githubusercontent.com/10414927/163197930-9229ee6c-63b9-433a-9048-e6d07904fb77.jpg)

as you can see I have SONOFF devices connected to each light circuit centrally so what would be your ideal solution, I use the AETOC quad switches for all my "buttons"

**Reply by joba-1** [reply]
I use domoticz myself and afaik Theo (the Tasmota author) uses that as well.
I have no big issues with it and when I evaluated it a few years ago I found it better than HA or OpenHAB in all categories I cared about (optics, stability, maintainability, apis, supported device types... or in other words effortlessness in all I wanted to achive since then). It is not perfect (nothing is) and that evaluation is old and may have changed. 
TLDR: worth a look :) 

And since you have a lot of devices, may I suggest to try out my domoticz plugin tasmoticz? It will pick up all tasmota devices (buttons and a lot of sensors) from their normal mqtt traffic, no manual effort at all.

**Reply by jamescatania** [reply]
@joba-1 Of course will do, will check out domoticz out for sure, thanks for the headsup

**Reply by jamescatania** [reply]
> Talk to whoever made that primitive Tasmota integration for Smartthings. A much better way of doing it would be to use MQTT, where Tasmota can automatically report status changes as it happens, no polling needed. And Tasmota is not optimized for the slower http requests anyway.
> 
> More generally, the extra power draw from constant commands to be processed may overtax the 3.3V power supply ability of some borderline devices, leading to them becoming "less responsive".

I disconnected the smartthings, but I'm still lagging as Tasmota keeps checking the connection

`11:31:31.771 WIF: Checking connection...
11:31:51.765 WIF: Checking connection...
11:32:09.380 WIF: Sending Gratuitous ARP
11:32:11.789 WIF: Checking connection...
11:32:31.787 WIF: Checking connection...
11:32:51.784 WIF: Checking connection...
11:33:09.373 WIF: Sending Gratuitous ARP
11:33:11.781 WIF: Checking connection...
11:33:31.785 WIF: Checking connection...
11:33:51.784 WIF: Checking connection...
11:34:09.381 WIF: Sending Gratuitous ARP
11:34:11.790 WIF: Checking connection...
11:34:31.800 WIF: Checking connection...
11:34:51.791 WIF: Checking connection...
11:35:09.392 WIF: Sending Gratuitous ARP
11:35:11.798 WIF: Checking connection...
11:35:31.800 WIF: Checking connection...
11:35:51.759 WIF: Checking connection...
11:36:09.392 WIF: Sending Gratuitous ARP
11:36:11.800 WIF: Checking connection...
11:36:17.422 RSL: STATE = {"Time":"2022-04-14T11:36:17","Uptime":"0T00:05:10","UptimeSec":310,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"ICARUS","BSSId":"74:AC:B9:D3:59:57","Channel":11,"Mode":"11n","RSSI":100,"Signal":-31,"LinkCount":1,"Downtime":"0T00:00:04"}}
11:36:31.798 WIF: Checking connection...
11:36:51.769 WIF: Checking connection...
11:37:09.416 WIF: Sending Gratuitous ARP
11:37:11.779 WIF: Checking connection...
11:37:31.780 WIF: Checking connection...
11:37:51.782 WIF: Checking connection...
11:38:09.378 WIF: Sending Gratuitous ARP
11:38:11.791 WIF: Checking connection...
11:38:31.753 WIF: Checking connection...
11:38:51.762 WIF: Checking connection...`

**Reply by jamescatania** [reply]
> What do you want Tasmota to do? From your logs, Tasmota is just answering what is being asked by HTTP.

I have switch off Smartthings completely and now TASMOTA keeps checking connection anyways

11:31:31.771 WIF: Checking connection...
11:31:51.765 WIF: Checking connection...
11:32:09.380 WIF: Sending Gratuitous ARP
11:32:11.789 WIF: Checking connection...
11:32:31.787 WIF: Checking connection...
11:32:51.784 WIF: Checking connection...
11:33:09.373 WIF: Sending Gratuitous ARP
11:33:11.781 WIF: Checking connection...
11:33:31.785 WIF: Checking connection...
11:33:51.784 WIF: Checking connection...
11:34:09.381 WIF: Sending Gratuitous ARP
11:34:11.790 WIF: Checking connection...
11:34:31.800 WIF: Checking connection...
11:34:51.791 WIF: Checking connection...
11:35:09.392 WIF: Sending Gratuitous ARP
11:35:11.798 WIF: Checking connection...
11:35:31.800 WIF: Checking connection...
11:35:51.759 WIF: Checking connection...
11:36:09.392 WIF: Sending Gratuitous ARP
11:36:11.800 WIF: Checking connection...
11:36:17.422 RSL: STATE = {"Time":"2022-04-14T11:36:17","Uptime":"0T00:05:10","UptimeSec":310,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"ICARUS","BSSId":"74:AC:B9:D3:59:57","Channel":11,"Mode":"11n","RSSI":100,"Signal":-31,"LinkCount":1,"Downtime":"0T00:00:04"}}
11:36:31.798 WIF: Checking connection...
11:36:51.769 WIF: Checking connection...
11:37:09.416 WIF: Sending Gratuitous ARP
11:37:11.779 WIF: Checking connection...
11:37:31.780 WIF: Checking connection...
11:37:51.782 WIF: Checking connection...
11:38:09.378 WIF: Sending Gratuitous ARP
11:38:11.791 WIF: Checking connection...
11:38:31.753 WIF: Checking connection...
11:38:51.762 WIF: Checking connection...

**Reply by sfromis** [reply]
Do not worry about that "Checking connection...". When you choose to run with log level 4, you get to see many details which are usually not interesting. For normal usage, I'd recommend log level 2 or maybe 3.

**Reply by jamescatania** [reply]
> Do not worry about that "Checking connection...". When you choose to run with log level 4, you get to see many details which are usually not interesting. For normal usage, I'd recommend log level 2 or maybe 3.

The initial issue was not with checking connection, whenever smartthings is on , keeps sending status commands through web, someone said the issue is the hongstat and how the tasmota connect is coded. Smarthings is sending STATUS commands to over 50 devices continously , every 5 seconds or so, "STATUS 8" and "STATUS 11". This is the real issue. For the life of me, not on any documentation I can find an interval setting or something of the sort, hence why i said in the beginning of this post, it is most probably a smartthings issue and not a tasmota issue.

Although I'm not trying to upgrade all the TASMOTA devices, however, even the web ui of the tasmota devices, most of the times is not loading, and I have to reload it multiple times for it to load up (eventually)

**Reply by sfromis** [reply]
My only point was that you should **always** see those "Checking connection..." if running with log level 4. You most probably did not do that all the time. I was responding to your "and now TASMOTA keeps checking connection anyways", which is not a "real difference", but due to you using a higher log level.

**Comment by ascillato** [comment, marked as answer]
The Wi-Fi Checking connection does not produce lag. The http status query from smarthings don't produce lag either. Where do you see a lag? In the webinterface? If so, may be it is just a Wi-Fi connection issue. If your RSSI is below 50%, your device can lag but for connectivity issues.

**Comment by Jason2866** [comment]
So many wifi devices so close together is maybe a problem.
